### PR TITLE
Tx signing: Increase encoded message length size to u16

### PR DESCRIPTION
**Pull request by CriesofCarrots**

### CODE REVIEW

File: src/signMessage.c [hunk 1]
```diff
@@ -64,8 +64,8 @@ void handleSignMessage(uint8_t p1, uint8_t p2, uint8_t *dataBuffer, uint16_t dat
     cx_ecfp_private_key_t privateKey;
     derive_private_key(&privateKey, derivationPath, derivationPathLength);
 
-    int messageLength = dataBuffer[0];
-    dataBuffer += 1;
+    int messageLength = (dataBuffer[0] << 8) + dataBuffer[1];
```
Comment (garious) [inline src/signMessage.c]: Big Endian, eh? Is that what others are doing for ADPU messages?

Comment (CriesofCarrots) [reply src/signMessage.c]: Just realized we are also size limited here: https://github.com/solana-labs/ledger-app-solana/blob/afb7eb2201161459dfa70c8e8777130002f354fc/src/main.c#L63 (`G_io_apdu_buffer[OFFSET_LC]`), so more work needed.

I haven't done a real exhaustive search of other ledger apps on the BE vs LE point (most are accepting data lengths <256), but we are using BE for the key derivation path already, a la Stellar: https://github.com/solana-labs/ledger-app-solana/blob/afb7eb2201161459dfa70c8e8777130002f354fc/src/getPubkey.c#L20

Comment (CriesofCarrots) [reply src/signMessage.c]: Okay, updated, and added long-message signing test.

**Comment by CriesofCarrots** [comment]
> Nice. Is that random data or an actual message?

It's random data. I'll update to actual solana messages when I incorporate Rust integration tests.

Also, back on endianness, I just confirmed the total APDU length is already encoded in big endian in the nanos-secure-sdk (https://github.com/LedgerHQ/nanos-secure-sdk/blob/1f2706941b68d897622f75407a868b60eb2be8d7/src/os_io_usb.c#L82), so while we don't have to go with BE here, I don't see much reason not to.

**Comment by garious** [comment]
Doesn't matter to me, but everything in the message is LE. Not obvious to me which is the side to stay consistent with. Up to you.

**Pull request merged**